src/Symfony/Component/Runtime/SymfonyRuntime.php

```diff
@@ -143,12 +147,23 @@ public function __construct(array $options = [])
 
         $options['error_handler'] ??= SymfonyErrorHandler::class;
 
+        $workerLoopMax = $options['worker_loop_max'] ?? $_SERVER['FRANKENPHP_LOOP_MAX'] ?? $_ENV['FRANKENPHP_LOOP_MAX'] ?? null;
+        if (null !== $workerLoopMax && (!\is_numeric($workerLoopMax) || $workerLoopMax <= 0)) {
+            throw new \LogicException(\sprintf('The "worker_loop_max" runtime option must be a positive integer, "%s" given.', $workerLoopMax));
+        }
+
+        $options['worker_loop_max'] = (int) $workerLoopMax ?? 500;
+
         parent::__construct($options);
     }
 
     public function getRunner(?object $application): RunnerInterface
     {
         if ($application instanceof HttpKernelInterface) {
+            if (filter_var($_SERVER['FRANKENPHP_WORKER'] ?? $_ENV['FRANKENPHP_WORKER'] ?? null, FILTER_VALIDATE_BOOLEAN, FILTER_NULL_ON_FAILURE) ?? false) {
```

Comment:
I don't think it's necessary to check for `$_ENV` or use `filter_var()`. IIRC Fr `$_SERVER` is always populated by FrankenPHP in worker mode, and the other SAPIs don't have any reason to use the `FRANKENPHP_` prefix.

Reply:
oh, this is not and env var, but something the sapi exposes? then indeed let's revert to the previous code

Reply:
Yes!